extensions/basic_auth/plugin.cc

```diff
@@ -0,0 +1,287 @@
+                      }
+                      request_path = path;
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request path.");
+              return false;
+            }
+            if (request_path == "" || match == "") {
+              return false;
+            }
+            if (!JsonArrayIterate(
+                    configuration, "request_methods",
+                    [&](const json& methods) -> bool {
+                      auto method = JsonValueAs<std::string>(methods);
+                      if (method.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        LOG_WARN("unexpected method");
+                        return false;
+                      }
+                      request_methods.push_back(
+                          std::string(method.first.value()));
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request methods.");
+              return false;
+            }
+            struct BasicAuthConfigRule rules;
+            if (!JsonArrayIterate(
+                    configuration, "credentials",
+                    [&](const json& credentials) -> bool {
+                      auto credential = JsonValueAs<std::string>(credentials);
+                      if (credential.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        LOG_WARN("unexpected credential");
+                        return false;
+                      }
+                      rules.encoded_credentials.insert(
+                          Base64::encode(credential.first.value().data(),
+                                         credential.first.value().size()));
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for credentials.");
+              return false;
+            }
+
+            if (match == "prefix") {
+              rules.pattern = Prefix;
+            } else if (match == "exact") {
+              rules.pattern = Exact;
+            } else if (match == "suffix") {
+              rules.pattern = Suffix;
+            }
+            rules.request_path = request_path;
+            for (auto& methods : request_methods) {
+              basic_auth_configuration_[methods].push_back(rules);
+            }
+            return true;
+          })) {
+    LOG_WARN("Failed to parse configuration information.");
+  }
+  return true;
+}
+
+FilterHeadersStatus PluginContext::credentialsCheck(
+    const PluginRootContext::BasicAuthConfigRule& rule,
+    const std::string& authorization_header) {
+  auto credential_iter = rule.encoded_credentials.find("");
+  auto credential_size = rule.encoded_credentials.size();
+  // Check if credential set is of of size 1 and its an empty string.
+  if (credential_size == 1 &&
```

Comment:
What is the purpose of this condition? Empty credential should be considered as an invalid configuration.